components/Menubar.tsx

```diff
@@ -1,15 +1,23 @@
 import { LogoutButton } from "./styled";
 import { useContext, useEffect } from "react";
 import { UserContext } from "../contexts/UserContext";
-import { setCurrentUser } from "../api/setCurrentUser";
+import { logout } from "../api/logout";
+import { useRouter } from "next/router";
 
 const Menubar = () => {
-  const { user } = useContext(UserContext);
+  const { user, setUser } = useContext(UserContext);
+  const router = useRouter();
+
+  const handleLogout = () => {
+    logout();
+    setUser(null);
+    router.push("/");
+  };

```

Comment:
I think we could move this function to the `UserContext` since it manages the `User`. That way, we just need to call the function (maybe `logoutUser`) to this component.

```suggestion
  const logoutUser = () => {
    logout();
    setUser(null);
    router.push("/");
  };

```

in MenuBar:
```
const { user, logoutUser } = useContext(UserContext);
<LogoutButton onClick={logoutUser}>Logout</LogoutButton>
    </div>
```